### PR TITLE
Improve example for fill_between

### PR DESCRIPTION
**Pull request by timhoffm**
## PR Summary

- More extensive (and hopefully instructive) descriptions.
- Plots better tailored to illustrate the descriptions.

<s>https://22159-1385122-gh.circle-artifacts.com/0/home/circleci/project/doc/build/html/gallery/lines_bars_and_markers/fill_between_demo.html#sphx-glr-gallery-lines-bars-and-markers-fill-between-demo-py</s>

https://22611-1385122-gh.circle-artifacts.com/0/home/circleci/project/doc/build/html/gallery/lines_bars_and_markers/fill_between_demo.html#sphx-glr-gallery-lines-bars-and-markers-fill-between-demo-py

### CODE REVIEW

File: examples/lines_bars_and_markers/fill_between_demo.py [hunk 1]
```diff
@@ -3,22 +3,27 @@
...
 x = np.arange(0.0, 2, 0.01)
 y1 = np.sin(2 * np.pi * x)
 y2 = 1.2 * np.sin(4 * np.pi * x)
 
-###############################################################################
-
 fig, (ax1, ax2, ax3) = plt.subplots(3, 1, sharex=True)
 
-ax1.fill_between(x, 0, y1)
+ax1.fill_between(x, y1)
```
Comment (anntzer) [inline examples/lines_bars_and_markers/fill_between_demo.py]: the y labels look really crammed, consider rewriting them to e.g. `from foo to bar` (shorter) or use titles or text(...).

File: examples/lines_bars_and_markers/fill_between_demo.py [hunk 2]
```diff
@@ -29,50 +34,116 @@
...
-ax1.fill_between(x, y1, y2, where=y2 <= y1,
-                 facecolor='red', interpolate=True)
-ax1.set_title('Now regions with y2>1 are masked')
+#
+# Example: Confidence bands
+# -------------------------
+# A common application for `~.axes.Axes.fill_between` is the indication of
+# confidence bands.
+#
+#  `~.axes.Axes.fill_between` uses the colors of the color cycle as the fill
```
Comment (anntzer) [inline examples/lines_bars_and_markers/fill_between_demo.py]: overindented paragraph

File: examples/lines_bars_and_markers/fill_between_demo.py [hunk 3]
```diff
@@ -29,50 +34,116 @@
...
 
 fig, ax = plt.subplots()
-y = np.sin(4 * np.pi * x)
+x = np.arange(0, 4 * np.pi, 0.01)
+y = np.sin(x)
 ax.plot(x, y, color='black')
 
-# use data coordinates for the x-axis and the axes coordinates for the y-axis
 import matplotlib.transforms as mtransforms
 trans = mtransforms.blended_transform_factory(ax.transData, ax.transAxes)
```
Comment (anntzer) [inline examples/lines_bars_and_markers/fill_between_demo.py]: this is just `ax.get_xaxis_transform()`, and then you can kill the import above and the reference to blended_transform_factory in the import list below.

File: examples/lines_bars_and_markers/fill_between_demo.py (hunk 1 above)
Comment (timhoffm) [reply examples/lines_bars_and_markers/fill_between_demo.py]: Done.

File: examples/lines_bars_and_markers/fill_between_demo.py [hunk 4]
```diff
@@ -3,76 +3,143 @@
...
+# therefore often a good practice to lighten the color by making the area
+# semi-transparent using *alpha*.
+
+# sphinx_gallery_thumbnail_number = 2
+
+N = 21
+x = np.linspace(0, 10, 11)
+y = [3.9, 4.4, 10.8, 10.3, 11.2, 13.1, 14.1,  9.9, 13.9, 15.1, 12.5]
+
+# fit a linear curve an estimate it's y-values and their error.
```
Comment (anntzer) [inline examples/lines_bars_and_markers/fill_between_demo.py]: its

File: examples/lines_bars_and_markers/fill_between_demo.py [hunk 5]
```diff
@@ -3,76 +3,143 @@
...
+#
+# References
+# """"""""""
+#
+# The use of the following functions, methods and classes is shown
+# in this example:
+
+import matplotlib
+matplotlib.axes.Axes.fill_between
+matplotlib.pyplot.fill_between
```
Comment (anntzer) [inline examples/lines_bars_and_markers/fill_between_demo.py]: also get_xaxis_transform, perhaps?

Comment (timhoffm) [reply examples/lines_bars_and_markers/fill_between_demo.py]: Sure.

**Comment by jklymak** [comment]
`AttributeError: module 'matplotlib.axes' has no attribute 'get_xaxis_transform'`

**Comment by timhoffm** [comment]
@jklymak Thanks for the reminder! Fixed.

**Comment by QuLogic** [comment]
@meeseeksdev backport to v3.1.x

**Pull request merged**